src/mesh/mesh_function.C

```diff
@@ -71,7 +71,18 @@ MeshFunction::MeshFunction (const EquationSystems & eqn_systems,
 {
 }
 
-
+MeshFunction::MeshFunction (const MeshFunction & mf):
+  FunctionBase<Number> (mf._master),
+  ParallelObject       (mf._eqn_systems),
+  _eqn_systems         (mf._eqn_systems),
+  _vector              (mf._vector),
+  _dof_map             (mf._dof_map),
+  _system_vars         (mf._system_vars),
+  _out_of_mesh_mode    (false)
+{
+  this->init();
```

Comment:
I'm not sure we want to call this unless ```mf.initialized()``` - someone might want to copy construct (or clone) a MeshFunction before the underlying Mesh is ready.  Although again this looks like there's a pre-existing bug - in our previous clone() code we checked initialized() before calling init(), but we then set_point_locator_tolerance() willy-nilly regardless of whether we know we've initialized _point_locator or not.